lib/ansible/module_utils/vmware.py

```diff
@@ -29,20 +30,42 @@ class TaskError(Exception):
     pass
 
 
-def wait_for_task(task):
+def wait_for_task(task, max_backoff=64, timeout=3600):
+    """Wait for given task using exponential back-off algorithm.
+
+    Args:
+        task: VMware task object
+        max_backoff: Maximum amount of sleep time in seconds
+        timeout: Timeout for the given task in seconds
+
+    Returns: True with result for successful task.
+    Raises: TaskError on failure
+    """
+    failure_counter = 0
+    start_time = time.time()
+
+    # List of generic errors
+    gen_errors = (
```

Comment:
I think this var needs a more specific name. Like `generic_vmw_errors`, if I understand correctly. Because now it's not clear for me what exactly you're generalizing.